File: 03_palindrome_check.cpp
#include <bits/stdc++.h>
using namespace std;

/*
Task 3: Check if a string is a palindrome.
Reads a line and prints "YES" if palindrome, otherwise "NO".
*/

int main()
{
    string s;
    if (!getline(cin, s))
        return 0;
    int i = 0, j = (int)s.size() - 1;
    while (i < j)
    {
        if (s[i] != s[j])
        {
            cout << "NO\n";
            return 0;
        }
        ++i;
        --j;
    }
    cout << "YES\n";
    return 0;
}

// Time Complexity: O(n)
// Space Complexity: O(1)

File: 03_palindrome_check.py
"""
Task 3: Check if a string is a palindrome.
Reads a line and prints "YES" if palindrome, otherwise "NO".
"""
import sys

def is_palindrome(s: str) -> bool:
    i, j = 0, len(s)-1
    while i < j:
        if s[i] != s[j]:
            return False
        i += 1; j -= 1
    return True

if __name__ == '__main__':
    s = sys.stdin.readline().rstrip('\n')
    print('YES' if is_palindrome(s) else 'NO')

# Time Complexity: O(n)
# Space Complexity: O(1)
